File: src/main/java/net/dontdrinkandroot/example/angularrestspringsecurity/repository/UserRepository.java
package net.dontdrinkandroot.example.angularrestspringsecurity.repository;

import net.dontdrinkandroot.example.angularrestspringsecurity.entity.User;
import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;
import org.springframework.data.rest.core.annotation.RestResource;

import java.util.Optional;

@RepositoryRestResource()
public interface UserRepository extends PagingAndSortingRepository<User, Long>
{
    @RestResource(exported = false)
    Optional<User> findByUsername(String username);
}

File: src/main/kotlin/net/dontdrinkandroot/example/angularrestspringsecurity/repository/UserRepository.kt
package net.dontdrinkandroot.example.angularrestspringsecurity.repository

import net.dontdrinkandroot.example.angularrestspringsecurity.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.rest.core.annotation.RepositoryRestResource
import org.springframework.data.rest.core.annotation.RestResource

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
@RepositoryRestResource
interface UserRepository : JpaRepository<User, Long> {

    @RestResource(exported = false)
    fun findByUsername(username: String): User?
}
